### PR TITLE
Include deposit status in Transaction exports

### LINKED ISSUE #2715: Add a "Deposit status" column to the transactions CSV export
Related to #2705

The transactions CSV export contains a "Deposit date" column though it's not clear whether this date refers to the estimated deposit date or the actual deposit date.

We should add a "Deposit status" column to indicate the current status of the deposit for each transaction.

This column should contain the deposit status as per our existing options - https://github.com/Automattic/woocommerce-payments/blob/develop/client/deposits/strings.js#L13

At the time of writing these are:

- Paid
- Pending
- In transit
- Canceled
- Failed
- Estimated

### PR DESCRIPTION
**Pull request by malithsen**
Fixes #2715

#### Changes proposed in this Pull Request
As mentioned in #2715, this adds a deposit_status column to Transaction exports. Additionally, this change replaces `client/deposits/strings.js` with a typescript file.
There's a separate server PR to expose deposit_status in the transaction response at 1385-gh-Automattic/woocommerce-payments-server

#### Testing instructions

* As an admin user with deposit records already available in Payments -> Deposits
* Go to Payments -> Transactions
* Click Download and open the exported csv file
* Notice the new Deposit Status column
* Ensure that deposit status is correct for the relevant transaction

-------------------

- [x] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

### CODE REVIEW

**Review by leonardola — CHANGES_REQUESTED** [review]
I am able to see the deposit_status inside the response.
I am able to download the CSV and see the deposit_status inside it.
I am able to see Estimated before the date inside Payments > Transactions.
I am able to click on the deposit link and see Estimated on the link.
I am able to see Estimated inside Payments > Deposits.

There is only a minor change about a duplicated status that I commented during review.

Other than that nice job. I am still reviewing the server PR

**Comment by leonardola** [comment]
Server PR is approved. After you solve the comment left during review you can merge it. I'll pre approve this

**Review by leonardola — APPROVED** [review]

File: client/data/transactions/hooks.ts [hunk 1]
```diff
@@ -30,6 +30,14 @@ export interface Transaction {
...
 	customer_currency: string;
 	deposit_id?: string;
+	deposit_status?:
+		| 'estimated'
+		| 'paid'
+		| 'pending'
+		| 'in_transit'
+		| 'canceled'
+		| 'failed'
+		| 'estimated';
```
[the remark this answers is not in the document]
Comment (malithsen) [reply client/data/transactions/hooks.ts:39]: Fixed in https://github.com/Automattic/woocommerce-payments/pull/3473/commits/994ca3926bbf9c303f10c2b32c8a1772f5370e22

**Comment by malithsen** [comment]
Thanks for the review @leonardola. Now that the server change is deployed, I'm merging this as well.

**Pull request merged**